winpr/libwinpr/clipboard/posix.c

```diff
@@ -1017,12 +1011,14 @@ static UINT posix_file_read_perform(struct posix_file* file, UINT32 size, BYTE**
 	return ERROR_READ_FAULT;
 }
 
-static UINT posix_file_read_close(struct posix_file* file)
+UINT posix_file_read_close(struct posix_file* file, BOOL force)
 {
 	if (file->fd < 0)
 		return NO_ERROR;
 
-	if (file->offset == file->size)
+	/* Always force close the file. Clipboard might open hundreds of files
+	 * so avoid caching to prevent running out of available file descriptors */
+	if ((file->offset >= file->size) || force || TRUE)
```

Comment:
I think the `|| TRUE` is redundant (you already check for `force`).
Other than that looks good.

Reply:
@kubistika no, I´ve force disabled this until we have determined such a fd cache will not always keep (lots of) handles open.